### PR TITLE
Switch to using new bootstrap script to initialize buildtools.

### PR DESCRIPTION
**Pull request by crummel**
- Move bootstrapping code from init-tools into run-build.
- Call the buildtools bootstrap script instead of doing our own bootstrapping so we don't end up with two different versions of the CLI.

Related to https://github.com/dotnet/buildtools/issues/918.

### CODE REVIEW

**Comment by crummel** [comment]
cc @weshaggard @chcosta

File: .cliversion [hunk 1]
```diff
@@ -0,0 +1 @@
+1.0.0-preview3-003708
```
Comment (weshaggard) [inline .cliversion]: Did you intend to change the version of the cli with this change?

File: run-build.ps1 [hunk 2]
```diff
@@ -30,6 +30,7 @@ if($Help)
 $env:CONFIGURATION = $Configuration;
```
Comment (livarcocc) [inline run-build.ps1]: Don't you need to do the same thing for the run-build.sh?

File: run-build.ps1 [hunk 3]
```diff
@@ -59,15 +60,41 @@ if (!(Test-Path $env:DOTNET_INSTALL_DIR))
...
+& $bootStrapperPath (Get-Location) $toolsLocalPath $env:DOTNET_INSTALL_DIR | Out-File (Join-Path (Get-Location) "bootstrap.log")
+if ($LastExitCode -ne 0)
+{
+    Write-Output "Boot-strapping failed with exit code $LastExitCode, see bootstrap.log for more information."
+    exit $LastExitCode
+}
+
+# create a junction to the shared FX version directory. this is
+# so we have a stable path to dotnet.exe regardless of version.
+$junctionTarget = Join-Path $env:DOTNET_INSTALL_DIR "shared\Microsoft.NETCore.App\$sharedFrameworkVersion"
```
Comment (crummel) [inline run-build.ps1]: Updated so bootstrap handles it all, removed.

File: run-build.ps1 [hunk 4]
```diff
@@ -59,15 +60,41 @@ if (!(Test-Path $env:DOTNET_INSTALL_DIR))
...
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+}
+
+# now execute it
+& $bootStrapperPath (Get-Location) $toolsLocalPath $env:DOTNET_INSTALL_DIR | Out-File (Join-Path (Get-Location) "bootstrap.log")
```
Comment (crummel) [inline run-build.ps1]: Yes, that's in https://github.com/dotnet/buildtools/pull/1107.

File: .cliversion (hunk 1 above)
Comment (crummel) [inline .cliversion]: I'm not sure where I got that from, it looks like the actual rel-1.0.0 version that we are fetching with the old script is 1.0.0-preview3-003786.  Updated.

File: run-build.ps1 (hunk 2 above)
Comment (crummel) [inline run-build.ps1]: There's currently no bootstrap.sh, I'll see if there's an existing issue on that.

File: run-build.ps1 (hunk 4 above)
Comment (weshaggard) [inline run-build.ps1]: Please pass the parameters by name so it is clear what each are for.

File: scripts/obtain/dotnet-install.ps1 [hunk 5]
```diff
@@ -59,7 +59,8 @@ param(
    [switch]$DryRun,
    [switch]$NoPath,
    [string]$AzureFeed="https://dotnetcli.azureedge.net/dotnet",
-   [string]$UncachedFeed="https://dotnetcli.blob.core.windows.net/dotnet"
+   [string]$UncachedFeed="https://dotnetcli.blob.core.windows.net/dotnet",
+   [string]$RuntimeVersionFile
```
Comment (weshaggard) [inline scripts/obtain/dotnet-install.ps1]: This should probably be a default parameter as well.

File: scripts/obtain/dotnet-install.ps1 [hunk 6]
```diff
@@ -366,6 +367,29 @@ function DownloadFile([Uri]$Uri, [string]$OutPath) {
     }
 }
 
+function Get-Runtime-Version-From-Dotnet-Package([System.IO.Compression.ZipArchive]$Zip) {
```
Comment (weshaggard) [inline scripts/obtain/dotnet-install.ps1]: Is there a reason this needs to be in this file? Can we not actually do it in the bootstrap script itself? Just find the highest version folder in under Microsoft.NETCore.App of the installed cli.

Comment (weshaggard) [inline scripts/obtain/dotnet-install.ps1]: I would much prefer that then reading and writing that value to another file.

File: run-build.ps1 (hunk 4 above)
Comment (crummel) [inline run-build.ps1]: Changed as suggested.

File: scripts/obtain/dotnet-install.ps1 (hunk 6 above)
Comment (crummel) [inline scripts/obtain/dotnet-install.ps1]: My concern there was that if there's a higher version installed than the consuming repo wants to use, we will use the higher version instead.  If that's not a problem, I'll change this to just use the highest version.

Comment (weshaggard) [inline scripts/obtain/dotnet-install.ps1]: I think we can give repo's an option to specify the version as a default parameter and if it isn't specified then you can find the highest version in the installed cli.

Comment (crummel) [inline scripts/obtain/dotnet-install.ps1]: Ok, I'll add that parameter and default to the highest version.

File: run-build.ps1 (hunk 2 above)
Comment (weshaggard) [inline run-build.ps1]: That's correct there isn't currently a bootstrap.sh and and I'm not aware of a tracking issue but I know it was on Joel's list in the past. It is definitely something we need.

**Comment by weshaggard** [comment]
Changes look good but we should get the bootstrap.sh online before merging.

**Comment by crummel** [comment]
@weshaggard I've updated to include the same build changes for Linux now that the bootstrapper is in.

**Comment by weshaggard** [comment]
LGTM but I don't work in this repo so I would prefer we get someone here to sign-off. @livarcocc

**Pull request merged**